TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
+        self.taObj = TAObj(self.taDB)  # creating TA object using TA in database.
+        tempUserForAdmin = User(email_address="admin@example.com")
+        tempAdmin = Administrator(user=tempUserForAdmin)
+        self.adminObj = AdminObj(tempAdmin)
+
+        self.courseDB = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+
+    # 1] TA w/ 1 course assignment
+    def test_1Crse1MaxCap(self):
+        TAToCourse.objects.create(ta=self.taDB, course=self.courseDB)  # assigning TA to course using db?
+        self.assertEquals(self.taObj.hasMaxAsgmts(), True,
+                          msg="TA has 1 max assignments & assigned 1 course: @ max cap")
+
+    # 2] TA w/ 0 course assignment
+    def test_0Crse1MaxCap(self):
+        self.assertEquals(self.taObj.hasMaxAsgmts(), False,
+                          msg="TA has 1 max assignments & not assigned 1 course: not @ max cap")
+
+    # 3] TA w/ max cap -> no max assign
+    def test_origMaxCapToNoAsgn(self):
+        TAToCourse.objects.create(ta=self.taDB, course=self.courseDB)
+        self.adminObj.removeCourse(CourseObj(self.courseDB))  # removing course should also remove this TA's assignment
+        self.assertEquals(self.taObj.hasMaxAsgmts(), False,
+                          msg="TA originally w/ assignment, removed, shouldn't be at max cap")
 
 
 class TestTAAssignTACourse(TestCase):  # Kiran
-    pass
+    taDB = None
+    courseDBList = list()
+    user = None  # for TA
+    taObj = None
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.taDB = TA.objects.create(user=self.user, max_assignments=2)  # max 2 assignment!
+        self.taObj = TAObj(self.taDB)  # creating TA object using TA in database
+
+        for i in [1, 2, 3]:
+            self.courseDBList.append(Course.objects.create(
+                course_id=100 + i,
+                semester='Fall 2023',
+                name='Introduction to Testing',
+                description='A course about writing tests in Django.',
+                num_of_sections=3,
+                modality='Online',
+                credits=4
+            ))
+
+    # 1] Success TA->Course
+    def test_ExistCourse(self):
+        self.taObj.assignTACourse(self.courseDBList[0])
+        self.assertIn(self.courseDBList[0], TAToCourse.objects.filter(course=self.courseDBList[0], ta=self.taDB).course
+                      , "Should have linked course and TA together")
+
+    # 2] Adding Course not existing in DB
+    def test_NotExistCourse(self):
```

Comment:
This test does not work because you are passing a course database obj to a method which probably should be taking a CourseObj object. Additionally, this test is the same as trying to add a TA to a non course. So you can just remove this test entirely

Reply:
I make that mistake a lot (sending in db object instead of the class obj to edit) - thank you. 

I don't see how this is similar to the TA adding to a non-course (this one would probably be a valueError because it has the proper argument type but invalid value - not saved to the db)